src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -99,14 +99,13 @@ def __init__(
         """
         self._flow_file = flow_file
         self._working_dir = Flow._resolve_working_dir(flow_file, working_dir)
-        if self._is_eager_flow_yaml():
-            self._is_dag_yaml_flow = False
-            self._program_language = FlowLanguage.Python
-        else:
+
+        self._is_eager_flow, self._program_language = self._check_eager_flow_and_language_from_yaml()
```

Comment:
@lumoslnt we don't have an official name for non-eager flow, so I think `_is_eager_flow` is a more clear name

Reply:
Good to me.